src/java/fr/paris/lutece/plugins/forms/web/entrytype/IEntryDataService.java

```diff
@@ -89,7 +90,7 @@
      *            the response to convert
      * @return the list of {@code String} objects
      */
-    List<String> responseToStrings( FormQuestionResponse formQuestionResponse );
+    Map<Integer, List<String>> responseToIterationsStrings( FormQuestionResponse formQuestionResponse );
```

Comment:
Does this break compatibility with plugins implementing IEntryDataService ? Is this absolutely necessary ? If so then you should change the major version of this plugin

Reply:
I will push other changes because what i pushed in this pull request isn't what it is expected. And indeed i realized it also included commits from other tickets in this pull request which is strange.
In order to resolve this, what can you suggest me to do :
- push my next changes in this Pull request ?
- create an other pull request (named  FORMS-828-2) ?

---

"_Does this break compatibility with plugins implementing IEntryDataService ? Is this absolutely necessary ? If so then you should change the major version of this plugin_" ?
-> IEntryDataService is contained in plugin-forms, but yes the service name seems to indicate it's also related to an other plugin (i don't know which plugin it is)

I'll try to not change this class if possible. If not i'll do it in the other plugin as you suggest. In what plugin is it ?

Reply:
Or i can just close this pull request and create a new one when i'll push

Reply:
yes please do

Reply:
> -> IEntryDataService is contained in plugin-forms, but yes the service name seems to indicate it's also related to an other plugin (i don't know which plugin it is)
> 
> I'll try to not change this class if possible. If not i'll do it in the other plugin as you suggest. In what plugin is it ?

you are changing an interface which might be implemented by other plugins. These plugins can be site specific, or in private repos. So you should really weight wether this change is absolutely necessary, and if it is, if you can provide a compatibility layer (with a default function for instance)

Reply:
> I will push other changes because what i pushed in this pull request isn't what it is expected

If you take another approach (what you did is indeed _not_ what the user asked for), you should start over, and not keep any of these commits.